base/linalg/triangular.jl

```diff
@@ -976,18 +976,71 @@ for (f, g) in ((:/, :A_rdiv_B!), (:A_rdiv_Bc, :A_rdiv_Bc!), (:A_rdiv_Bt, :A_rdiv
     end
 end
 
+function ^(A::UpperTriangular, p::Integer)
+    if p < 0
+        return UpperTriangular(Base.power_by_squaring(inv(A), -p))
+    else
+        return UpperTriangular(Base.power_by_squaring(A, p))
+    end
+end
+# Complex matrix power for upper triangular factor, see:
+#   Higham and Lin, "A Schur-Padé algorithm for fractional powers of a Matrix",
+#     SIAM J. Matrix Anal. & Appl., 32 (3), (2011) 1056–1078.
+#   Higham and Lin, "An improved Schur-Padé algorithm for fractional powers of
+#     a matrix and their Fréchet derivatives", SIAM. J. Matrix Anal. & Appl.,
+#     34(3), (2013) 1341–1360.
+function powm{T<:Union{Float64,Complex{Float64}}}(A0::UpperTriangular{T}, p::Real)
+    theta = [1.53e-5, 2.25e-3, 1.92e-2, 6.08e-2, 1.25e-1, 2.03e-1, 2.84e-1]
+    n = chksquare(A0)
+
+    A, m, s = invsquaring(A0,theta)
+    A = I - A
+
+    # Compute accurate diagonal of I - T
+    sqrt_diag!(A0, A, s)
+    for i = 1:n
+        A[i,i] = -A[i,i]
+    end
+    # Compute the Padé approximant
+    mpm = 2 * m
+    c = 0.5 * (p - m) / (mpm - 1)
+    S = c * A
+    for j = m-1:-1:1
```

Comment:
It seems like this loop makes a lot of copies of matrices.  Can't the scaling of `S` be done in-place?